### PR TITLE
Dev Mode - Social Card Editing - Download Links

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR touches on a few different areas that we are needing for further development.

- Dev Mode: Using `npm run start:dev` can now startup the server in dev mode. Further details in the PR readme.
- Social Card Editing: When in dev mode the new API Endpoint `/dev/image/packages/:packageName` is now exposed to allow contributors to edit and view the raw HTML that creates our social cards.
- Download Links: The new API Endpoint `/download` allows easy linking to our Cirrus CI binaries. Using this endpoint and it's two query parameters you can easily link to the most recent Cirrus CI build available. Further details on the query parameters and how to craft a link for this endpoint are in the `/docs/download_links.md` file.

Please feel free to ask for any clarifications on this PR. Thanks!

### CODE REVIEW

**Comment by confused-Techie** [comment]
@Daeraxa would you think this one is good to merge at this point then?

**Comment by Daeraxa** [comment]
Cards yes but I completely forgot about the download endpoints part. Let me have a quick play with that.

**Comment by confused-Techie** [comment]
> Cards yes but I completely forgot about the download endpoints part. Let me have a quick play with that.

Sure thing, take your time

**Comment by Daeraxa** [comment]
The downloads work really nicely, it will be super easy to integrate that into the downloads page, great work.

I tested all the binary download combinations, the only ones that didn't work are the two mac .zips which is to be expected until that gets resolved.

File: docs/download_links.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- `mac_zip`
+- `mac_zip_blockmap`
+- `mac_dmg`
+- `mac_dmg_blockmap`
+- `linux_appimage`
+- `linux_tar`
+- `linux_rpm`
+- `linux_deb`
+
+Full Table:
```
Comment (Daeraxa) [inline docs/download_links.md:38]:
```suggestion
Example: https://web.pulsar-edit.dev/download?os=windows&type=windows_portable

Full Table:
```
I know it is kind of obvious but if it saves somebody 30 seconds of googling how to make a url query then I think it is worth it.

Comment (confused-Techie) [reply docs/download_links.md:38]: Good idea on that one, like we always preach, lowest barrier to entry at all times

**Review by Daeraxa — APPROVED** [review]

**Comment by confused-Techie** [comment]
Thanks for the approval @Daeraxa just saw it, so I'll go ahead and merge, then update prod

**Pull request merged**